internal/federationdomain/oidc/oidc_test.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2024 the Pinniped contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package oidc
+
+import (
+	"testing"
+	"time"
+
+	"github.com/ory/fosite"
+	"github.com/stretchr/testify/require"
+
+	"go.pinniped.dev/internal/federationdomain/clientregistry"
+)
+
+func TestDefaultLifespans(t *testing.T) {
+	c := DefaultOIDCTimeoutsConfiguration()
```

Comment:
I'm a little confused on the naming here. I know the names weren't introduced in this PR, but I see a function with `Timeouts` that has `Lifespans` and `Lifetimes`.

Also, the object returned by this function has more fields than these five.

Reply:
Yeah, maybe confusing that things related to tokens are called "lifespans" and things related to session storage are called "lifetimes". The token lifespans are `Durations` and the storage lifespans are functions that return `Durations`.

The name "lifespan" for tokens is copied from fosite. `fosite.Config` has fields with names like these: https://github.com/vmware-tanzu/pinniped/blob/v0.29.0/internal/federationdomain/oidc/oidc.go#L135-L138 which is what our fields eventually get assigned into.

For whatever historical reason, when we developed the session storage code we called everything "lifetimes" for that code (see `crud.go` along with all the callers of `crud.New`).

We can refactor the names if you'd like.

Reply:
not necessary for this PR